node/heartbeat/monitor.go

```diff
@@ -416,10 +423,12 @@ func (m *Monitor) computeInactiveHeartbeatMessages() {
 // GetHeartbeats returns the heartbeat status
 func (m *Monitor) GetHeartbeats() []PubKeyHeartbeat {
 	m.mutHeartbeatMessages.Lock()
-	status := make([]PubKeyHeartbeat, len(m.heartbeatMessages))
-	idx := 0
+	status := make([]PubKeyHeartbeat, 0, len(m.heartbeatMessages))
 	for k, v := range m.heartbeatMessages {
-		v.updateMutex.Lock()
+		if m.shouldSkipValidator(v) {
+			continue
```

Comment:
delete(m.heartbeat, k) ?
A node should simply not care about an unresponsive observers.

Reply:
yes, good point. It should be removed from the heartBeatMessages. If it is online it will be added again.